OSDOCS-4763: fix microshift configuration example (#54335)
openshift/openshift-docs: modules/microshift-config-yaml.adoc

:_content-type: CONCEPT
[id="microshift-config-yaml_{context}"]
= Using a YAML configuration file
= Configuring {product-title}

Configure your {product-title} using a YAML file. The values specified in your configuration file are used when you execute a command.

> // Q: can these values be overriden with CLI flags?

Author: Technically, yes, but we do not want to support that long-term and will likely remove it soon.

Reviewer: "Using a YAML configuration file" is Docs-style correct. "Configuring MicroShift" is vague and hard to find, especially if other methods are added later.

Author: Restored

{product-title} searches for a configuration file in the user-specific directory, `~/.microshift/config.yaml`, then the system-wide `/etc/microshift/config.yaml` directory. Create the configuring file, specifying any settings that should override the defaults, before starting MicroShift.

[id="microshift-yaml-default_{context}"]
== Default settings
If you do not create a `config.yaml` file, the default values are used.

> // Q: what creates the default yaml?

Author: There is a default file in /etc/microshift/config.yaml.example that comes from the RPM package. Nothing reads that, it's just meant as a help to the user. If no config file exists, the defaults are built into the code.

Author: Done

> // Q: can default file be modified? (instead of making a new one)

Author: You only have to include the settings you want to change in the config file, but you have to put them in a file.

network:
  clusterNetwork:
    - cidr: 10.42.0.0/16 <2>
  serviceNetwork:
    - 10.43.0.0/16 <3>
  serviceNodePortRange: 30000-32767 <4>
node:
  hostnameOverride: "" <5>
  nodeIP: "" <6>
apiServer:
  subjectAltNames: [] <7>
debugging:
  logLevel: "Normal" <8>
----

[IMPORTANT]
<1> Base domain of the cluster. All managed DNS records will be sub-domains of this base.
<2> A block of IP addresses from which Pod IP addresses are allocated.
<3> A block of virtual IP addresses for Kubernetes services.
<4> The port range allowed for Kubernetes services of type NodePort.
<5> The name of the node, defaults to hostname.
<6> The IP address of the node, defaults to IP of the default route.
<7> Subject Alternative Names for apiserver certificates.
<8> Log verbosity. Valid values are Normal, Debug, Trace, or TraceAll.

[NOTE]
====
The configuration file must be located at the user-specific directory, `~/.microshift/config.yaml`, and the system-wide `/etc/microshift/config.yaml` directory. An existing user-specific `config.yaml` takes precedence.
MicroShift only reads the configuration file on startup. Restart MicroShift after changing any configuration settings to have them take effect.
====

> //Q: can this be done with a command, or is this an instance where user goes into the directory and creates a text file?

Author: They just need to create the text file. This may be a case where less is more, because there are zillions of ways for them to create that file so just describing what it looks like and where it should be ought to be enough.

Reviewer: Per the original engineering doc, this config file has to be in the directory listed in line 35. Should we be deleting line 35? That is the location where the file must ought to be, unless I am misunderstanding.

Author: That content was repeated from line 9, so I removed the duplication.

[id="microshift-yaml-format-example_{context}"]
== Formatting a customized YAML file

.Procedure
//Q: need explicit steps
. Create a YAML file with the following formatting:
+
[source,yaml]
----
cluster: ""
  clusterCIDR: ""
  serviceCIDR: ""
  serviceNodePortRange: ""
  dns: ""
  domain: ""
  url: ""
nodeIP: ""
nodeName: ""
logVLevel: ""
----

. Enter valid values.

> //Q: most docsets contain a table of valid entry types, noting whether required or optional

Author: Added

Reviewer: lgtm

> //Q: does the user need to enter every value? will any default?

Author: All of them have defaults. Not all of the defaults are necessarily what someone would want, but they do work.

> //Q: how does the user check that this procedure has been done properly?

Author: That would depend on which config values they are changing, and could lead to a lot of explaining. Let's look at that for 4.13.

> //Q: Can we give a sample use-case set of values in an example YAML? see below

Author: I added the defaults to the example above.

[id="microshift-config-yaml_{context}"]
= Using a YAML configuration file

> {product-title} searches for a configuration file in the user-specific directory, `~/.microshift/config.yaml`, then the system-wide `/etc/microshift/config.yaml` directory. Create the configuring file, specifying any settings that should override the defaults, before starting MicroShift.

Reviewer 2: This is a concept module, but the sentence "Create the configuring file..." makes it sound more like a procedure, so this is a rewording suggestion:
```suggestion
{product-title} searches for a configuration file in the user-specific directory, `~/.microshift/config.yaml`, then the system-wide `/etc/microshift/config.yaml` directory. You must create the configuration file and specify any settings that should override the defaults before starting {product-title}.
```

[id="microshift-yaml-default_{context}"]
== Default settings

> If you do not create a `config.yaml` file, the default values are used. Create the configuration file, specifying any settings that should override the defaults, before starting {product-title}.

Reviewer 2: Similar to the above comment:
```suggestion
If you do not create a `config.yaml` file, the default values are used. The following example configuration contains the default settings. You must change any settings that should override the defaults before starting {product-title}.
```

.Default YAML file example
[source,yaml]
----
dns:
  baseDomain: microshift.example.com <1>
network:
  clusterNetwork:
    - cidr: 10.42.0.0/16 <2>
  serviceNetwork:
    - 10.43.0.0/16 <3>
  serviceNodePortRange: 30000-32767 <4>
node:
  hostnameOverride: "" <5>
  nodeIP: "" <6>
apiServer:
  subjectAltNames: [] <7>
debugging:
  logLevel: "Normal" <8>
----

> <1> Base domain of the cluster. All managed DNS records will be sub-domains of this base.

Reviewer 2: ISG nit:
```suggestion
<1> Base domain of the cluster. All managed DNS records will be subdomains of this base.
```

<2> A block of IP addresses from which Pod IP addresses are allocated.
<3> A block of virtual IP addresses for Kubernetes services.
<4> The port range allowed for Kubernetes services of type NodePort.

> <5> The name of the node, defaults to hostname.

Reviewer 2: ```suggestion
<5> The name of the node. The default value is the hostname.
```

> <6> The IP address of the node, defaults to IP of the default route.

Reviewer 2: ```suggestion
<6> The IP address of the node. The default value is the IP address of the default route.
```

> <7> Subject Alternative Names for apiserver certificates.

Reviewer 2: ```suggestion
<7> Subject Alternative Names for API server certificates.
```

> <8> Log verbosity. Valid values are Normal, Debug, Trace, or TraceAll.

Reviewer 2: Since these are YAML config values, I think they need to be formatted like other [API object references](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references):
```suggestion
<8> Log verbosity. Valid values for this field are `Normal`, `Debug`, `Trace`, or `TraceAll`.
```

[NOTE]
====

> MicroShift only reads the configuration file on startup. Restart MicroShift after changing any configuration settings to have them take effect.

Reviewer 2: I'm guessing `{product-title}` resolves to MicroShift:
```suggestion
{product-title} only reads the configuration file on startup. Restart {product-title} after changing any configuration settings to have them take effect.
```
